compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/Patterns.cpp

```diff
@@ -174,6 +175,75 @@ struct ConvertTensorCastPattern : public OpRewritePattern<tensor::CastOp> {
   }
 };
 
+struct ConvertTensorConcatPattern : public OpRewritePattern<tensor::ConcatOp> {
+  using OpRewritePattern<tensor::ConcatOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(tensor::ConcatOp concatOp,
+                                PatternRewriter &rewriter) const override {
+    if (concatOp->getParentOfType<IREE::Flow::DispatchRegionOp>() ||
+        concatOp->getParentOfType<IREE::Flow::DispatchWorkgroupsOp>()) {
+      return failure();
+    }
+    if (concatOp.getDim() != 0) {
```

Comment:
We could also support cases where the outer most dims are all 1 in the future.

Reply:
I would think that would be "foldUnitDims" on concat ops...